### PR TITLE
Intake

### PR DESCRIPTION
**Pull request by IHaveBadAim1**
## Describe your changes

- [ ] Bug fix
- [X] New feature

Explain what you did:
I made intake.
> ...

## Issue ticket (add number)

> https://github.com/FRC2706/2024-2706-Robot-Code/issues/...

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [X] I have tested thoroughly.
- [X] I have written documentation and explained about my changes to the best of my ability.

### CODE REVIEW

**Comment by weizhang-2706** [comment]
Need to resolve conflicts.

File: src/main/java/frc/robot/Config.java [hunk 1]
```diff
@@ -248,4 +250,9 @@ public static final class AutoConstants {
         new TrapezoidProfile.Constraints(
             kMaxAngularSpeedRadiansPerSecond, kMaxAngularSpeedRadiansPerSecondSquared);
   }
+
+    public static final class Intake {
+      public static final int INTAKE = -1;
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/Config.java]: This could be put into class CANID, with robotSpecific.

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

**Review by GabbyGenereux — CHANGES_REQUESTED** [review]
Please merge into `Integration` instead of main.

File: src/main/java/frc/robot/commands/MakeIntakeMotorSpin.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        intakeSubsystem = IntakeSubsystem.getInstance();
+
+        if (intakeSubsystem != null) {
+            addRequirements(intakeSubsystem);
+        }
+
+    } 
+
+
+    public MakeIntakeMotorSpin(CommandXboxController driver, int value, int value2) {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/MakeIntakeMotorSpin.java]: you can clean up this.

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

**Review by weizhang-2706 — APPROVED** [review]

File: src/main/java/frc/robot/robotcontainers/NewRobotContainer.java [hunk 3]
```diff
@@ -30,6 +34,7 @@ public class NewRobotContainer extends RobotContainer {
   private final CommandXboxController operator = new CommandXboxController(1);
 
   private final SwerveSubsystem s_Swerve = SwerveSubsystem.getInstance();
+  private final IntakeSubsystem intake = IntakeSubsystem.getInstance();
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/robotcontainers/NewRobotContainer.java]: You can remove this line. intake subsystem will be created when the command is scheduled first time.

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/robotcontainers/NewRobotContainer.java [hunk 4]
```diff
@@ -55,8 +60,15 @@ public NewRobotContainer() {
...
   private void configureButtonBindings() {
     
-
+    CommandXboxController driver = new CommandXboxController(0);
+    CommandXboxController operator = new CommandXboxController(1);
     /* Driver Controls */
+    driver.a().whileTrue(new MakeIntakeMotorSpin(0.6, 0));
+    //driver.a()
+    //.whileTrue(Commands.run(()->{intake.setVoltage(10);}, intake))
+    //.whileFalse(Commands.run(()->{intake.setVoltage(0);}, intake));
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/robotcontainers/NewRobotContainer.java]: clean up these three lines.

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

**Review by ErikCald — APPROVED** [review]

**Review by weizhang-2706 — APPROVED** [review]

File: src/main/java/frc/robot/commands/MakeIntakeMotorSpin.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
+package frc.robot.commands;
```
Comment (GabbyGenereux) [inline src/main/java/frc/robot/commands/MakeIntakeMotorSpin.java:1]: Please rename this class to something like "IntakeCommand"

File: src/main/java/frc/robot/robotcontainers/BeetleContainer.java [hunk 6]
```diff
@@ -14,6 +14,7 @@
 import frc.robot.commands.BlingCommand;
 import frc.robot.commands.BlingCommand.BlingColour;
 import frc.robot.subsystems.DiffTalonSubsystem;
+import frc.robot.commands.MakeIntakeMotorSpin;
```
Comment (GabbyGenereux) [inline src/main/java/frc/robot/robotcontainers/BeetleContainer.java:17]: remove this

File: src/main/java/frc/robot/commands/MakeIntakeMotorSpin.java (hunk 5 above)
Comment (IHaveBadAim1) [reply src/main/java/frc/robot/commands/MakeIntakeMotorSpin.java:1]: ok

File: src/main/java/frc/robot/robotcontainers/BeetleContainer.java (hunk 6 above)
Comment (IHaveBadAim1) [reply src/main/java/frc/robot/robotcontainers/BeetleContainer.java:17]: ok

**Pull request merged**